flow/skia_gpu_object_unittests.cc

```diff
@@ -31,19 +32,133 @@ class TestSkObject : public SkRefCnt {
   fml::TaskQueueId* dtor_task_queue_id_;
 };
 
-TEST_F(SkiaGpuObjectTest, UnrefQueue) {
-  fml::RefPtr<fml::TaskRunner> task_runner = CreateNewThread();
-  fml::RefPtr<SkiaUnrefQueue> queue = fml::MakeRefCounted<SkiaUnrefQueue>(
-      task_runner, fml::TimeDelta::FromSeconds(0));
+// Death tests and threads do not get along.  This fixture gives access to the
+// current thread context but doesn't start any other threads.
+using SkiaGpuObjectDeathTest = ThreadTest;
+
+// This fixture starts a 2nd thread by default.  The |SkiaUnrefQueue| uses the
+// 2nd thread to perform unref() operations on its data items.
+class SkiaGpuObjectTest : public SkiaGpuObjectDeathTest {
+ public:
+  SkiaGpuObjectTest()
+      : unref_task_runner_(CreateNewThread()),
+        unref_queue_(fml::MakeRefCounted<SkiaUnrefQueue>(
+            unref_task_runner(),
+            fml::TimeDelta::FromSeconds(0))),
+        delayed_unref_queue_(fml::MakeRefCounted<SkiaUnrefQueue>(
+            unref_task_runner(),
+            fml::TimeDelta::FromSeconds(3))) {}
+  ~SkiaGpuObjectTest() override = default;
+
+  fml::RefPtr<fml::TaskRunner> unref_task_runner() {
+    return unref_task_runner_;
+  }
+  fml::RefPtr<SkiaUnrefQueue> unref_queue() { return unref_queue_; }
+  fml::RefPtr<SkiaUnrefQueue> delayed_unref_queue() {
+    return delayed_unref_queue_;
+  }
+
+ private:
+  fml::RefPtr<fml::TaskRunner> unref_task_runner_;
+  fml::RefPtr<SkiaUnrefQueue> unref_queue_;
+  fml::RefPtr<SkiaUnrefQueue> delayed_unref_queue_;
+};
+
+TEST_F(SkiaGpuObjectDeathTest, CreateObjectWithNoQueueDies) {
```

Comment:
This restriction is being relaxed in a patch under review. Depending on the order in which they land, you may need to remove the same. Just a heads up in case a rebase causes failures.